packages/edit-site/src/components/navigation-sidebar/navigation-panel/menus/content-categories.js

```diff
@@ -1,23 +1,65 @@
 /**
  * WordPress dependencies
  */
-import { __experimentalNavigationMenu as NavigationMenu } from '@wordpress/components';
+import {
+	__experimentalNavigationMenu as NavigationMenu,
+	__experimentalNavigationItem as NavigationItem,
+} from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
-import NavigationEntityItems from '../navigation-entity-items';
 import { MENU_CONTENT_CATEGORIES, MENU_ROOT } from '../constants';
+import ContentNavigationItem from '../content-navigation-item';
+import SearchResults from '../search-results';
+import useDebouncedSearch from '../use-debounced-search';
 
 export default function ContentCategoriesMenu() {
+	const {
+		search,
+		searchQuery,
+		onSearch,
+		isDebouncing,
+	} = useDebouncedSearch();
+
+	const categories = useSelect(
+		( select ) =>
+			select( 'core' ).getEntityRecords( 'taxonomy', 'category', {
+				search: searchQuery,
+			} ),
+		[ searchQuery ]
+	);
+
+	const isLoading = ! search && categories === null;
+	const shouldShowLoadingForDebouncing = search && isDebouncing;
+	const showLoading = isLoading || shouldShowLoadingForDebouncing;
+
 	return (
 		<NavigationMenu
 			menu={ MENU_CONTENT_CATEGORIES }
 			title={ __( 'Categories' ) }
 			parentMenu={ MENU_ROOT }
+			hasSearch={ true }
+			onSearch={ onSearch }
+			search={ search }
```

Comment:
Should we also be adding the `isSearchDebouncing` prop here?

Reply:
Ooops, forgot about categories 😄 Pushed a commit to add this.